packages/editor/src/lib/editor/managers/LicenseManager.ts

```diff
@@ -26,38 +26,57 @@ interface ValidLicenseKeyResult {
 }
 
 export class LicenseManager {
-	private publicKey = '3UylteUjvvOL4nKfN8KfjnTbSm6ayj23QihX9TsWPIM='
+	private publicKey: string
 	private isTest: boolean
-	constructor() {
+	constructor(testPublicKey?: string) {
 		this.isTest = typeof process !== 'undefined' && process.env.NODE_ENV === 'test'
+		this.publicKey = testPublicKey || '3UylteUjvvOL4nKfN8KfjnTbSm6ayj23QihX9TsWPIM='
 	}
-	private extractLicense(licenseKey: string): LicenseInfo {
-		const base64License = util.decodeBase64(licenseKey)
+	private async extractLicenseKey(licenseKey: string): Promise<LicenseInfo> {
+		const [data, signature] = licenseKey.split('.')
+		const [prefix, encodedData] = data.split('/')
 
-		const decoded = nacl.sign.open(base64License, util.decodeBase64(this.publicKey))
+		if (prefix !== 'tldraw') {
+			throw new Error(`Unsupported prefix '${prefix}'`)
+		}
+
+		const publicCryptoKey = await importPublicKey(this.publicKey)
 
-		if (!decoded) {
-			throw new Error('Invalid license key')
+		try {
+			await crypto.subtle.verify(
+				{
+					name: 'ECDSA',
+					hash: { name: 'SHA-384' },
+				},
+				publicCryptoKey,
+				new Uint8Array(str2ab(signature)),
```

Comment:
It kept throwing an error saying that these and other arguments were not an Array Buffer. I checked them with ```instanceof``` , and saw that they definitely were. At some point I tried this and it worked. Not sure why it helps!

Reply:
haha, yeah, it's kind of annoying - i think we can just do `as Uint8Array` instead of actually doing wrapping it a `new Uint8Array` but pull me aside if it's still giving you trouble